tests/test_black.py

```diff
@@ -1352,6 +1352,40 @@ def test_parse_pyproject_toml(self) -> None:
         self.assertEqual(config["exclude"], r"\.pyi?$")
         self.assertEqual(config["include"], r"\.py?$")
 
+    def test_parse_pyproject_toml_project_metadata(self) -> None:
+        for test_toml, expected in [
+            ("only_black_pyproject.toml", ["py310"]),
+            ("only_metadata_pyproject.toml", ["py37"]),
+            ("neither_pyproject.toml", None),
+            ("both_pyproject.toml", ["py310"]),
+        ]:
+            test_toml_file = THIS_DIR / "data" / "project_metadata" / test_toml
+            config = black.parse_pyproject_toml(str(test_toml_file))
+            self.assertEqual(config.get("target_version"), expected)
+
+    def test_infer_target_version(self) -> None:
+        for version, expected in [
+            ("3.6", TargetVersion.PY36),
+            ("3.11.0rc1", TargetVersion.PY311),
+            (">=3.10", TargetVersion.PY310),
+            (">3.6,<3.10", TargetVersion.PY37),
+            ("==3.8.*", TargetVersion.PY38),
+            ("==3.*", TargetVersion.PY33),
+            (">=3.8.6", TargetVersion.PY38),
+            ("> 3.7.4, != 3.8.8", TargetVersion.PY37),
+            (">3.7,!=3.8", TargetVersion.PY39),
```

Comment:
```suggestion
            (">3.7,!=3.8", TargetVersion.PY39),
            ("!=3.3,!=3.5", TargetVersion.PY35),
```

Reply:
Your test case doesn't work (returns TargetVersion.PY34). I think you meant `"!=3.3,!=3.4"`. I added this test case in a new commit.